src/bridge/error/error.cc

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/bridge/error/error.h"
+
+#include <stdio.h>
+
+#include <openssl/err.h>
+
+#include "src/backing/status/status_code.h"
+#include "src/bridge/error/error_strings.h"
+#include "src/bridge/error/function_code.h"
+
+namespace kmsengine {
+namespace bridge {
+namespace error {
+
+// OpenSSL engines request a library code using ERR_get_next_error_library(3)
+// with which they can register engine-specific, human-readable error strings
+// with OpenSSL.
+static const int library_code = ERR_get_next_error_library();
+
+void LoadErrorStringsIntoOpenSSL() {
+  ERR_load_strings(library_code,
+                   const_cast<ERR_STRING_DATA*>(kFunctionStrings));
```

Comment:
As gross as it would be, it might make sense to just make `kFunctionStrings` and `kReasonStrings` non-const. Otherwise, we could crash if OpenSSL tries to modify the arrays.

Reply:
Ah, they can't be `const` because OpenSSL will modify the tables anyways to patch in the library code...yeah, I guess they have to be non-const here.

There's an undocumented `ERR_load_strings_const` function that it seems the OpenSSL library uses internally (see this [commit](https://mta.openssl.org/pipermail/openssl-commits/2017-June/014925.html)). I'm trying to find an example of an engine that uses it to see if it's something we could use here. It appears that the semantics of `ERR_load_strings_const` are that the library code has to be packed in by the engine code (which is different from `ERR_load_strings` as mentioned above).

Reply:
Never mind. It looks like even if you use `ERR_load_strings_const`, you still have to `const_cast` anyways since there is no `ERR_unload_strings_const` version. Thus it looks like it can't be const no matter what to avoid the `const_cast`.